crates/cast/src/starknet_commands/declare.rs

```diff
@@ -116,15 +111,12 @@ pub async fn declare(
         })
         .ok_or(anyhow!("Failed to find starknet_artifacts.json file"))?;
 
-    let starknet_artifacts = match std::fs::read_to_string(starknet_artifacts) {
-        Ok(content) => content,
-        Err(err) => {
-            return Err(anyhow!(
-                "Failed to read starknet_artifacts.json contents: {}",
-                err
-            ))
-        }
-    };
+    let starknet_artifacts = std::fs::read_to_string(&starknet_artifacts).with_context(|| {
+        format!(
+            "Failed to read {} contents",
+            starknet_artifacts.into_os_string().into_string().unwrap()
```

Comment:
not sure we should use unwrap here, what if this fails?

Reply:
Would doing this be okay?
```rust
            "Failed to read {:?} contents",
            starknet_artifacts
```

Reply:
yes, I think this is acceptable :+1: